src/index.ts

```diff
@@ -69,7 +69,7 @@ export function Serializable(baseClassName?: string): Function {
 /**
  * Function to deserialize json into a class
  */
-export function deserialize(json: any, type: any): any {
+export function deserialize<T>(json: any, type: new (..._: Array<any>) => T): T {
```

Comment:
Why do you need to use `new (..._: Array<any>) => T`, just `new () => T` works?

Reply:
`new () => T` will work only if type T has a default constructor.

Otherwise,

```
Argument of type 'typeof T' is not assignable to parameter of type 'new () => T'. ts(2345)
```

Reply:
Could you send me the test you did to get the error message.

Like this I could add it to my own tests please :)

Reply:
```
    it('should accept a non-default(has parameter(s) on) constructor', () => {
      @Serializable()
      class NonDefault {
        constructor(
          @JsonProperty()
          public readonly parameter1: string
        ) {}
      }

      // won't compile if you use 'new () => T'
      deserialize({ parameter1: 'parameter1' }, NonDefault)
    })
```